gateway/middleware.go

```diff
@@ -48,8 +49,6 @@ type TykMiddleware interface {
 	GetSpec() *APISpec
 
 	Init()
-	SetName(string)
-	SetRequestLogger(*http.Request)
```

Comment:
Both are being removed from the interface as they are BaseMiddleware interface functions.